src/config/loadGitHubAppsFromEnvironment.ts

```diff
@@ -53,42 +80,70 @@ export class GitHubAppsRegistry {
   // API for payload (JSON struct coming from GitHub)
 
   loadFromPayload(payload) {
-    // Soon we aim to support multiple orgs!
-    const org = '__tmp_org_placeholder__'; // payload?.organization?.login;
-    if (!org) {
-      throw new Error(`Could not find an org in ${JSON.stringify(payload)}.`);
+    // Org slug is differently accessed in org-scoped APIs vs. repo-scoped APIs
+    let org: string;
+    if (payload.organization) {
+      org = payload.organization.login;
+    } else if (payload.repository?.owner?.type === 'Organization') {
+      org = payload.repository.owner.login;
+    } else {
+      throw new Error(
+        `Could not find an org in '${JSON.stringify(
+          payload.organization
+        )}' or '${JSON.stringify(payload.repository?.owner)}'.`
+      );
     }
     return this.load(org);
   }
 }
 
 export function loadGitHubAppsFromEnvironment(env) {
   const configs = new GitHubAppsConfigHelper();
-  let config;
-
-  if (env.GH_APP_IDENTIFIER && env.GH_APP_SECRET_KEY) {
-    // Collect config by (proleptic) envvar number. Once we have GH_APP_1_FOO
-    // this will make more sense. We'll collect shtuff in config and then
-    // instantiate a GitHubApp once all config has been collected for each
-    // (once we've made a full pass through process.env).
-
-    config = configs.forNumber(1);
-    config.org = '__tmp_org_placeholder__';
-    config.auth = {
-      appId: Number(env.GH_APP_IDENTIFIER),
-      privateKey: env.GH_APP_SECRET_KEY.replace(/\\n/g, '\n'),
-    };
-    config.project = {
-      node_id: env.ISSUES_PROJECT_NODE_ID || 'PVT_kwDOABVQ184AOGW8',
-      product_area_field_id:
-        env.PRODUCT_AREA_FIELD_ID || 'PVTSSF_lADOABVQ184AOGW8zgJEBno',
-      status_field_id: env.STATUS_FIELD_ID || 'PVTSSF_lADOABVQ184AOGW8zgI_7g0',
-      response_due_date_field_id:
-        env.RESPONSE_DUE_DATE_FIELD_ID || 'PVTF_lADOABVQ184AOGW8zgLLxGg',
-    };
+
+  for (const [envvar, value] of Object.entries(env)) {
+    // Find app configuration in env, grouping by number (GH_APP_1_FOO).
+
+    const m = envvar.match(/^GH_APP_(\d+)_([A-Z_]+)$/);
+    if (m === null) {
+      continue;
+    }
+
+    const n = parseInt(m[1], 10);
```

Comment:
Well, so far Sentry has acquired two other companies in 15 years ... though it's accelerating! :o) I think we can live with this for a while and revisit when it becomes a pain point. Maybe we'll have a web UI by then and we can configure through that.

Reply:
> Is the idea that if I have multiple apps, I'd have N `GH_APP_N_ORG_SLUG`, and so on for each other env var, for each app?

But yes, that's the idea. And the point about acquisitions is that we have one app per org, and we (roughly) have one org per brand ~= company acquired.

Reply:
> we could just have `GH_APP_ORG_SLUGS` and then have the RHS be comma separated, then at runtime check to make sure each `GH_APP_*` field has the same number of entries

Also, I did consider this, but I feel that this is less scalable, in that it would be harder to visually inspect and modify envvars of this form. Imagine a comma-separated list of 20 apps, and you want to make a modification to number 13, but all 20 of them look like `PVT_af1728e2bf9b7133` or whatever. So dense, so error-prone. I think the present system scales better up to ~20 or so orgs/apps. By the time we get there I believe our situation will be different enough that we should evaluate available options then.